### PR TITLE
Metadata reserves 8 bits for global properties

### PR DESCRIPTION
**Pull request by mejango**

### CODE REVIEW

File: contracts/JBETHPaymentTerminal.sol [hunk 1]
```diff
@@ -64,25 +64,28 @@ contract JBETHPaymentTerminal is JBPayoutRedemptionPaymentTerminal {
...
     @param _to The address to which the transfer should go.
     @param _amount The amount of the transfer, as a fixed point number with the same number of decimals as this terminal.
   */
   function _transferFrom(
-    address,
+    address _from,
     address payable _to,
     uint256 _amount
   ) internal override {
+    _from; // Prevents unused var compiler and natspec complaints.
```
Comment (drgorillamd) [inline contracts/JBETHPaymentTerminal.sol:76]: \o/

File: contracts/JBFundingCycleStore.sol [hunk 2]
```diff
@@ -384,15 +385,15 @@ contract JBFundingCycleStore is IJBFundingCycleStore, JBControllerUtility {
...
+    if (!_isApproved(_projectId, _baseFundingCycle))
       // If it hasn't been approved, set the ID to be the funding cycle it's based on,
       // which carries the latest approved configuration.
-      _currentConfiguration = _getStructFor(_projectId, _currentConfiguration).basedOn;
+      _baseFundingCycle = _getStructFor(_projectId, _baseFundingCycle.basedOn);
 
-    // Determine the funding cycle to use as the base.
-    JBFundingCycle memory _baseFundingCycle = _getStructFor(_projectId, _currentConfiguration);
+    // The configuration can't be the same as the base configuration.
+    if (_baseFundingCycle.configuration == _configuration) revert NO_SAME_BLOCK_RECONFIGURATION();
```
Comment (drgorillamd) [inline contracts/JBFundingCycleStore.sol:396]: nice! that was a nasty bug in reconfig e2e!

File: contracts/JBETHPaymentTerminal.sol (hunk 1 above)
Comment (mejango) [reply contracts/JBETHPaymentTerminal.sol:76]: image
makes this much better ^^

File: hardhat.config.js [hunk 3]
```diff
@@ -65,7 +65,7 @@ module.exports = {
       optimizer: {
         enabled: true,
         // https://docs.soliditylang.org/en/v0.8.10/internals/optimizer.html#:~:text=Optimizer%20Parameter%20Runs,-The%20number%20of&text=A%20%E2%80%9Cruns%E2%80%9D%20parameter%20of%20%E2%80%9C,is%202**32%2D1%20.
-        runs: 1000000,
+        runs: 1000000
```
Comment (drgorillamd) [inline hardhat.config.js:68]: this thing doesn't make any sense

**Review by drgorillamd — APPROVED** [review]
lfg to the moon 1000x

File: hardhat.config.js (hunk 3 above)
Comment (mejango) [reply hardhat.config.js:68]: I think the move is to deploy at 10000, and in the readme note that tests need to be run at 1000000.

Meanwhile I've opened an issue with hardhat here https://github.com/NomicFoundation/hardhat/issues/2657

**Pull request merged**